src/test/java/com/saucelabs/bamboo/sod/action/BuildConfiguratorTest.java

```diff
@@ -129,6 +137,18 @@ public void sshDisabled() throws Exception {
         assertTrue("Sauce tunnels not empty", getTunnelMap().isEmpty());
     }
 
+    @Test
+    public void sshEnabledWithUniqueKey() throws Exception {
+        buildDefinition.getCustomConfiguration().put(SODKeys.SSH_ENABLED_KEY, "true");
+        buildDefinition.getCustomConfiguration().put(SODKeys.SSH_USE_GENERATED_TUNNEL_ID, "true");
+        buildDefinition.getCustomConfiguration().put(SODKeys.ENABLED_KEY, "true");
+        CustomVariableContext customVariableContext = mock(CustomVariableContext.class);
```

Comment:
Can this be new'd instead of mocked? since its spring I'm guessing its a lot of work and you have to find the impl class anyways.